app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -26,4 +26,25 @@ def google_oauth2
       redirect_to new_user_session_path
     end
   end
+
+  def google_billing
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      @user.update(authentication_token: Devise.friendly_token(32))
+      @user.generate_access_token
+      # update a user's FireCloud status
+      @user.delay.update_firecloud_status
```

Comment:
Fair point, although I think that deserves it's own PR as it touches a ton of code.  My one caveat to that is that while the service/application is now called Terra, the orchestration API is still api.firecloud.org, and many of their internal admin accounts still reference that domain.

Reply:
Agreed.  Ideally we would suggest to relevant upstream teams that those domains, etc. get renamed to Terra equivalents like api.terra.bio, then (after renames are done) update our code.

It might be worth raising if we're ever asked how upstream FireCloud / Terra APIs could be refined -- a future opportunity rather than an action item.